File: library/src/main/java/org/crispy_fish/filetype/classdef/SimpleStringSplittingClassDefLineReader.java
package org.crispy_fish.filetype.classdef;

import java.util.*;

public class SimpleStringSplittingClassDefLineReader implements ClassDefLineReader<String> {

    private final List<String> fields;

    public SimpleStringSplittingClassDefLineReader() {
        this.fields = new ArrayList<>();
    }

    SimpleStringSplittingClassDefLineReader(List<String> fields) {
        this.fields = fields;
    }

    public void setClassDefLine(String line) {
        fields.clear();
        String[] fieldsArray = line.split("\t");
        Collections.addAll(fields, fieldsArray);
    }

    public String getClassAbbreviation() {
        return fields.get(ClassDefIndices.CLASS_ABBREVIATION);
    }

    public String getPax() {
        return fields.get(ClassDefIndices.PAX);
    }

    public String getMysteryIndicatorAtIndex2() {
        return null;
    }

    public String getClassName() {
        return fields.get(ClassDefIndices.CLASS_NAME);
    }

    public String getMysteryIndicatorAtIndex4() {
        return null;
    }

    public String getMysteryIndicatorAtIndex5() {
        return null;
    }

    public String getMysteryIndicatorAtIndex6() {
        return null;
    }

    public String getCategory() {
        return fields.get(ClassDefIndices.CATEGORY);
    }


}

File: library/src/main/kotlin/org/coner/crispy_fish/filetype/classdef/SimpleStringSplittingClassDefLineReader.kt
package org.coner.crispy_fish.filetype.classdef

class SimpleStringSplittingClassDefLineReader(fields: MutableList<String>? = null) : ClassDefLineReader {

    var line: String? = null
        set(value) {
            fields.clear()
            if (value == null || value.isBlank()) return
            val fieldsArray = value.split("\t".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            fields.addAll(fieldsArray)
            field = value
        }
    private val fields = fields ?: mutableListOf()

    override val classAbbreviation: String
        get() = fields[ClassDefIndices.CLASS_ABBREVIATION]

    override val pax: String
        get() = fields[ClassDefIndices.PAX]

    override val mysteryIndicatorAtIndex2: String
        get() = throw UnsupportedOperationException()

    override val className: String
        get() = fields[ClassDefIndices.CLASS_NAME]

    override val mysteryIndicatorAtIndex4: String
        get() = throw UnsupportedOperationException()

    override val mysteryIndicatorAtIndex5: String
        get() = throw UnsupportedOperationException()

    override val mysteryIndicatorAtIndex6: String
        get() = throw UnsupportedOperationException()

    override val category: String
        get() = fields[ClassDefIndices.CATEGORY]

}
